### ISSUE TITLE
Oauth error failed_grant_with_invalid_scopes

### ISSUE THREAD

**Issue by ilrein** [Shopify/shopify_app#1839]
# Issue summary

Before opening this issue, I have:

- [x ] Upgraded to the latest version of the package
  - `shopify_app` version:
  - Ruby version:
  - Operating system:
- [ ] Set `log_level: :debug` [in my configuration](https://github.com/Shopify/shopify-api-ruby#setup-shopify-context), if applicable
- [ ] Found a reliable way to reproduce the problem that indicates it's a problem with the package
- [ ] Looked for similar issues in this repository
- [ ] Checked that this isn't an issue with a Shopify API
  - If it is, please create a post in the [Shopify community forums](https://community.shopify.com/c/partners-and-developers/ct-p/appdev) or report it to [Shopify Partner Support](https://help.shopify.com/en/support/partners/org-select)

I have created a fresh install. I am able to render the page successfully. When I try to install the shop, I get this:

<img width="854" alt="Screen Shot 2024-05-02 at 4 31 55 PM" src="https://github.com/Shopify/shopify_app/assets/5523014/06588c16-cb74-4ed3-af03-aecc4d1e3b12">

## Expected behavior

I left the defaults of:

```

ShopifyApp.configure do |config|
  config.application_name = "My Shopify App"
  config.old_secret = ""
  config.scope = "read_products" # Consult this page for more scope options:
                                  # https://help.shopify.com/en/api/getting-started/authentication/oauth/scopes
```

## Actual behavior

But it seems to have an issue with the scope.

## Steps to reproduce the problem

I ran all the steps to start a new project found in the README.

## Debug logs

I do see this on the failed request:

```
Started POST "/login" for 198.98.126.168 at 2024-05-02 16:40:13 -0400
Cannot render console from 198.98.126.168! Allowed networks: 127.0.0.0/127.255.255.255, ::1
Processing by ShopifyApp::SessionsController#create as HTML
  Parameters: {"authenticity_token"=>"[FILTERED]", "shop"=>"easy-tiles-easy-life.myshopify.com"}
Redirected to https://admin.shopify.com/store/easy-tiles-easy-life/oauth/install?client_id=82b530a33880214df01aeb372ca5a890
Completed 302 Found in 2ms (ActiveRecord: 0.0ms | Allocations: 610)
```

**Comment by hidavidday** [comment]
same issue is happening for me. Fresh install and all URLs are configured properly.

**Comment by mur-wtag** [comment]
@zzooeeyy I am also encountered with same issue and I triggered `dubug` mode and here is the log:

```ruby
 Shop Pluck (0.3ms)  SELECT "shops"."shopify_domain" FROM "shops"
19:37:07 │ web-backend      │   ↳ app/services/misc_shop_service.rb:10:in `allowed_origins'
19:37:07 │ web-backend      │ Started POST "/login" for 103.197.153.39 at 2024-06-12 19:37:07 +0600
19:37:07 │ web-backend      │ Cannot render console from 103.197.153.39! Allowed networks: 127.0.0.0/127.255.255.255, ::1
19:37:07 │ web-backend      │ Processing by ShopifyApp::SessionsController#create as HTML
19:37:07 │ web-backend      │   Parameters: {"authenticity_token"=>"[FILTERED]", "shop"=>"sofenx-dev.myshopify.com"}
19:37:07 │ web-backend      │ [ ShopifyApp | DEBUG | Shop Not Found ] Starting OAuth - Redirecting to Shopify managed install
19:37:07 │ web-backend      │ Redirected to https://admin.shopify.com/store/sofenx-dev/oauth/install?client_id=cfb75542ffbcafdac9ac271d68c583c5
19:37:07 │ web-backend      │ Completed 302 Found in 2ms (ActiveRecord: 0.0ms | Allocations: 717)

```

**[ ShopifyApp | DEBUG | Shop Not Found ] Starting OAuth - Redirecting to Shopify managed install**

**Comment by mur-wtag** [comment]
sorry I've forgot to mention here that I've resolved the issue. I did some debug and found in my `.toml` file there is

```
use_legacy_install_flow = true
```
I have found if we are using `new_embedded_auth_strategy`, we have to remove ☝️ this like from the `.toml` file.

After removing this line from `.toml` file it's working perfectly!

**Comment by andrewchen5678** [comment]
how did you initialize your app, on partner site maunally, through command line, or through shopify plus? I followed the README's instruction and it seems to suggest to initialize an app manually on partner site first.

**Comment by krishnadiamesso** [comment]
@andrewchen5678 Yeahhh creating the app from the CLI works for me as well. but there are a lot of things there I don't need or don't want to use. so wanted to go with a rails app from scratch.
will dig a bit more. worst comes to worst I will go the cli route.

**Comment by mzrnsh** [comment]
I followed the steps in the gem's README and got this error as well.

I used the manual option on partners site.

No shopify.app.toml file was created after I ran the generator. If this file is required, I hope the README gets a section explaining this.

**Comment by zzooeeyy** [comment]
Hey all! The reason why the CLI works and the generator doesn't work is because the `shopify_app` gem enabled a [new installation + authorization method](https://github.com/shopify/shopify_app?tab=readme-ov-file#new-embedded-app-authorization-strategy-token-exchange). 

Previously, your app (behind the scenes and handled in the gem) would need to redirect the user to shopify `admin/oauth/authorize` to initiate the installation process, and shopify redirects the user back to your app's callback path once the merchant grants permission to install your app. The gem then goes and retrieves an access token which your app can use to make API calls to that shop's resources. This process happens every-time an access token expires, invalidated, or you need to acquire extra scopes.

With the new **Shopify managed installation and token exchange authorization flow,** you declare the access scopes that your app requires in the `toml` file and deploy that to Shopify. So when a merchant tries to install your app, Shopify knows what access scopes to request for, and **Shopify will handle the installation** instead of redirecting to your app to make another redirect to install. And once your app is installed, and Shopify admin loads your app. `shopify_app` gem's token exchange logic will exchange that user's active id token for an access token.This is the best way to handle installation and token acquisition since it involves way less redirects. [You can see the videos here comparing the performance improvement](https://github.com/Shopify/shopify_app/pull/1834).

## Long story short, 
The generator doesn't configure shopify managed install for your app, but creating the app through the CLI does, that's why it has different behaviours.

#### There are a few options:
1. Configure your `.toml` to enable [Shopify managed install through the CLI](https://shopify.dev/docs/apps/build/authentication-authorization/app-installation#shopify-managed-installation) 

OR

2. Disable the new auth flow by changing (not recommended)
```
# config/initializers/shopify_app.rb
ShopifyApp.configure do |config|
  #.....
  config.new_embedded_auth_strategy = false
  ...
end
```

**Comment by farooki** [comment]
I have followed all the instructions given in shopify_app gem Readme and facing the similar issue 
"Oauth error failed_grant_with_invalid_scopes: The application could not be installed due to invalid scopes requested.
"

I created the application directly from shopify partner account rather than cli. anyone knows how can I fix it?

**Comment by danielfriis** [comment]
I appreciate the work you are all doing on this but I agree with @mzrnsh. For the Rails community, it seems like a bit of a mess. I'm hitting roadblock after roadblock and it feels way more complicated than it has to be.

You should hit the reset button and make a proper, simple solutions for vanilla Rails projects.

I'm writing this with love. It's simply because I _want_ to develop for Shopify, but it's just not that straightforward.

**Comment by danielfriis** [comment]
@zzooeeyy I see that this is "Closed as not planned". Could you at least update the documentation to reflect the new reality? (again, writing this with nothing but love, but I simply don't understand how to get up and running).
